libraries/AP_NavEKF2/AP_NavEKF2_core.h

```diff
@@ -1171,6 +1174,11 @@ class NavEKF2_core
     AP_HAL::Util::perf_counter_t  _perf_FuseOptFlow;
     AP_HAL::Util::perf_counter_t  _perf_test[10];
 
+    // earth field from WMM tables
+    bool have_table_earth_field;   // true when we have initialised table_earth_field_ga
```

Comment:
As a general rule, all new class variables should be explicitly reset in NavEKF2_core::InitialiseVariables() unless the initial value is truly a 'don't care'

Reply:
fixed